File: c/cz/mg/vulkan/PFN_vkCreateSampler.c
#include <vulkan/vulkan.h>
#include <jni.h>
#include <stdlib.h>
#include <string.h>

jlong jniPointerToLong(void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_vk_PFNvkCreateSampler_call(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pCreateInfo, jlong pAllocator, jlong pSampler, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkCreateSampler f = (PFN_vkCreateSampler)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkSamplerCreateInfo*)jniLongToPointer(pCreateInfo)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator)),
        ((VkSampler*)jniLongToPointer(pSampler))
    );
}

File: cz/mg/vulkan/PFNvkCreateSampler.java
package cz.mg.vulkan;

public class PFNvkCreateSampler extends VkFunctionPointer {
    public PFNvkCreateSampler() {
    }

    public PFNvkCreateSampler(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkCreateSampler(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }


    public PFNvkCreateSampler(long value) {
        setValue(value);
    }

    public PFNvkCreateSampler(VkInstance instance) {
        super(instance, new VkString("vkCreateSampler"));
    }

    public void call(VkDevice device, VkSamplerCreateInfo pCreateInfo, VkAllocationCallbacks pAllocator, VkSampler pSampler, VkResult rval){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddress(), pCreateInfo != null ? pCreateInfo.getVkAddress() : VkPointer.NULL, pAllocator != null ? pAllocator.getVkAddress() : VkPointer.NULL, pSampler != null ? pSampler.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getNullAddress());
    }

    private static native void call(long vkaddress, long device, long pCreateInfo, long pAllocator, long pSampler, long rval);
}
